SpatialGDK/Source/SpatialGDKFunctionalTests/Private/SpatialFunctionalTest.cpp

```diff
@@ -586,38 +608,24 @@ void ASpatialFunctionalTest::SetupClientPlayerRegistrationFlow()
 	));
 }
 
-void ASpatialFunctionalTest::CrossServerRegisterAutoDestroyActor_Implementation(AActor* ActorToAutoDestroy)
-{
-	RegisterAutoDestroyActor(ActorToAutoDestroy);
-}
-
-void ASpatialFunctionalTest::ServerRegisterAutoDestroyActor_Implementation(AActor* ActorToAutoDestroy)
+void ASpatialFunctionalTest::DeleteActorsRegisteredForAutoDestroy()
 {
-	CrossServerRegisterAutoDestroyActor(ActorToAutoDestroy);
-}
-
-void ASpatialFunctionalTest::MulticastAutoDestroyActors_Implementation(const TArray<AActor*>& ActorsToDestroy)
-{
-	FString DisplayName = LocalFlowController ? LocalFlowController->GetDisplayName() : TEXT("UNKNOWN");
-	if (!HasAuthority()) // Authority already handles it in Super::FinishTest
+	// Delete actors marked for auto destruction
+	if (UWorld* World = GEngine->GetWorldFromContextObject(GetWorld(), EGetWorldErrorMode::LogAndReturnNull))
 	{
-		for (AActor* Actor : ActorsToDestroy)
+		for (TActorIterator<AActor> It(World, AActor::StaticClass()); It; ++It)
 		{
-			if (IsValid(Actor))
-			{				
-				UE_LOG(LogSpatialGDKFunctionalTests, Display, TEXT("%s trying to delete actor: %s ; result now would be: %s"), *DisplayName, *Actor->GetName(), Actor->Role == ROLE_Authority ? TEXT("SUCCESS") : TEXT("FAILURE"));
-				Actor->SetLifeSpan(0.01f);
-			}
-		}
-	}
-	else
-	{
-		for (AActor* Actor : ActorsToDestroy)
-		{
-			if (IsValid(Actor))
+			AActor* FoundActor = *It;
+			UActorComponent* AutoDestroyComponent = FoundActor->FindComponentByClass<USpatialFunctionalTestAutoDestroyComponent>();
+			if (AutoDestroyComponent != NULL)
```

Comment:
Keep in mind that in c++ you should use nullptr instead of NULL, I think the C null is frowned upon.

Reply:
Ok, changed to nullptr.